src/groups/group.jl

```diff
@@ -1050,3 +1049,8 @@ function group_exp!(G::MultiplicationGroup, q, X)
     X isa Union{Number,AbstractMatrix} && return copyto!(q, exp(X))
     return error("group_exp! not implemented on $(typeof(G)) for vector $(typeof(X)) and element $(typeof(q)).")
 end
+
+function group_log!(G::MultiplicationGroup, X, q)
+    q isa Union{Number,AbstractMatrix} && return copyto!(X, log_safe(q))
+    return error("group_log! not implemented on $(typeof(G)) for element $(typeof(q)) and vector $(typeof(X)).")
```

Comment:
If I do restrict the arguments, then whenever anyone wants to overload `group_log!` downstream generically, they will need to implement two methods instead of one to avoid ambiguity. This approach reduces code duplication.

Reply:
I see, OK.